File: client/iOS/chalkak/chalkak/Features/Record/Components/RecordWeekdayHeader.swift
import SwiftUI

/// 일~토 요일 헤더. Android `RecordWeekdayHeader`와 맞춘다(caption, textMuted, 균등폭 중앙정렬).
/// 화면 좌우 패딩은 호출부가 관리한다.
struct RecordWeekdayHeader: View {
    @Environment(\.chalkakTheme) private var theme

    private static let weekdays = ["일", "월", "화", "수", "목", "금", "토"]

    var body: some View {
        HStack(spacing: theme.spacing.none) {
            ForEach(Self.weekdays, id: \.self) { weekday in
                Text(weekday)
                    .font(theme.typography.caption)
                    .foregroundStyle(theme.colors.textMuted)
                    .multilineTextAlignment(.center)
                    .frame(maxWidth: .infinity)
            }
        }
        .accessibilityHidden(true)
    }
}

#Preview("Record Weekday Header", traits: .sizeThatFitsLayout) {
    RecordWeekdayHeader()
        .padding(.horizontal, 20)
        .background(ChalkakTheme.light.colors.background)
        .chalkakTheme(.light)
}

File: client/android/app/src/main/java/com/stonefive/chalkak/feature/record/component/RecordWeekdayHeader.kt
package com.stonefive.chalkak.feature.record.component

import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import com.stonefive.chalkak.core.designsystem.theme.ChalkakTheme

private val Weekdays = listOf("일", "월", "화", "수", "목", "금", "토")

@Composable
fun RecordWeekdayHeader(modifier: Modifier = Modifier) {
    Row(modifier = modifier) {
        Weekdays.forEach { weekday ->
            Text(
                text = weekday,
                color = ChalkakTheme.colors.textMuted,
                style = ChalkakTheme.typography.caption,
                textAlign = TextAlign.Center,
                modifier = Modifier.weight(1f),
            )
        }
    }
}

@Preview(showBackground = true, widthDp = 402)
@Composable
private fun RecordWeekdayHeaderPreview() {
    ChalkakTheme {
        RecordWeekdayHeader(
            modifier = Modifier
                .fillMaxWidth()
                .padding(horizontal = 20.dp),
        )
    }
}
